### PR TITLE
Add namespaceSelector for spire-controller-manager

### PR DESCRIPTION
**Pull request by KenHuffmanAtNice**
Signed-off-by: Ken Huffman <Ken.Huffman@niceincontact.com>

### CODE REVIEW

File: charts/spire/charts/spiffe-csi-driver/templates/daemonset.yaml [hunk 1]
```diff
@@ -68,7 +76,7 @@ spec:
           args: [
             "-csi-address", "/spiffe-csi/csi.sock",
             "-kubelet-registration-path", "/var/lib/kubelet/plugins/csi.spiffe.io/csi.sock",
-            "-health-port", "9809"
+            "-health-port", {{ .Values.nodeDriverRegistrar.healthPort | quote }}
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-csi-driver/templates/daemonset.yaml:79]: What is the value of being able to change this? It just makes the helm chart complexer for the end user if there is no specific usecase to change this.

File: charts/spire/charts/spiffe-csi-driver/values.yaml [hunk 2]
```diff
@@ -46,6 +46,9 @@ securityContext:
 nodeSelector:
   kubernetes.io/arch: amd64
 
+affinity: {}
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-csi-driver/values.yaml:49]: Could you elaborate on the usecase for affinity on Daemonsets? What would be a practical example on when to use that?

File: charts/spire/charts/spiffe-csi-driver/templates/daemonset.yaml (hunk 1 above)
Comment (KenHuffmanAtNice) [reply charts/spire/charts/spiffe-csi-driver/templates/daemonset.yaml:71]: We use the Amazon EFS CSI Driver and it's daemonset uses this port for its healthPort (https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/charts/aws-efs-csi-driver/values.yaml#L124).

This _may_ be a non-issue if we can use the namespaceSelector to exclude it.

**Review by marcofranssen — APPROVED** [review]

**Comment by marcofranssen** [comment]
@KenHuffmanAtNice Please resolve the DCO.

```shell
git rebase main --sign-off
git push --force-with-lease
```

**Comment by KenHuffmanAtNice** [comment]
@marcofranssen Done. It is has been stuck on this for the past day:
> **Test** Expected — Waiting for status to be reported

**Pull request merged**